main.py

```diff
@@ -2,107 +2,102 @@
+    stats = []
 
-                stats = []
+    for answer in answer_iterator:
+        stats.append([answer._numeric._int_value, answer._owner._value])
 
-                for answer in answer_iterator:
-                    stats.append([answer._numeric._int_value, answer._owner._value])
+    stats.sort(reverse = True)
 
-                stats.sort(reverse = True)
-
-                print("\nTop 3 countries with most Vaccinations:")
-                for i in range(3):
-                    print(stats[i][1], ": ", stats[i][0])
+    print("\nTop 3 countries with most Vaccinations:")
+    for i in range(3):
+        print(stats[i][1], ": ", stats[i][0])
 
 def vaccineByAge(lower_limit, upper_limit):
-    
-    with TypeDB.core_client("localhost:1729") as client:
-        with client.session("vaccination", SessionType.DATA) as session:
-            with session.transaction(TransactionType.READ) as read_transaction:
 
-                answer_iterator = read_transaction.query().match_group_aggregate(f'match $p isa person, has age >= {lower_limit}, has age <= {upper_limit}; $m isa manufacturer, has vaccine_name $v; $vcc (person: $p, manufacturer: $m) isa vaccinator; get $p, $v; group $v; count;')
+    answer_iterator = read_transaction.query().match_group_aggregate(f'match $p isa person, has age >= {lower_limit}, has age <= {upper_limit}; $m isa manufacturer, has vaccine_name $v; $vcc (person: $p, manufacturer: $m) isa vaccinator; get $p, $v; group $v; count;')
 
-                print(f"\nVaccines given to age group {lower_limit} - {upper_limit} years:")
+    print(f"\nVaccines given to age group {lower_limit} - {upper_limit} years:")
 
-                for answer in answer_iterator:
-                    print(answer._owner._value, ": ", answer._numeric._int_value)
+    for answer in answer_iterator:
+        print(answer._owner._value, ": ", answer._numeric._int_value)
 
 def vaccineByCountry(country):
 
-    with TypeDB.core_client("localhost:1729") as client:
-        with client.session("vaccination", SessionType.DATA) as session:
-            with session.transaction(TransactionType.READ) as read_transaction:
-
-                country_iterator = read_transaction.query().match('match $p isa person, has country $c; get $c;')
+    country_iterator = read_transaction.query().match('match $p isa person, has country $c; get $c;')
 
-                countries = []
-                for country_pres in country_iterator:
-                    name = country_pres.get("c")
-                    countries.append(name.get_value())
+    countries = []
+    for country_pres in country_iterator:
+        name = country_pres.get("c")
+        countries.append(name.get_value())
 
-                if country not in countries:
-                    print("No entries for the given country!")
-                    return
+    if country not in countries:
+        print("No entries for the given country!")
+        return

```

Comment:
Perhaps I wasn't entirely clear - I don't think you need this initial query *at all* - your second query (the one you store in `answer_iterator`), in the case where no such country is present, will simply not return any results - and you can handle that case in the response from that query.